distributed/cluster_dump.py

```diff
@@ -57,3 +60,219 @@ def writer(state: dict, f: IO):
         # Write from a thread so we don't block the event loop quite as badly
         # (the writer will still hold the GIL a lot though).
         await to_thread(writer, state, f)
+
+
+def load_cluster_dump(url: str) -> dict:
+    """Loads a cluster dump from a disk artefact
+
+    Parameters
+    ----------
+    url : str
+        Name of the disk artefact. This should have either a
+        `.msgpack.gz` or `yaml` suffix, depending on the dump format.
+
+    Returns
+    -------
+    state : dict
+        The cluster state at the time of the dump.
+    """
+    if url.endswith(".msgpack.gz"):
+        mode = "rb"
+        reader = msgpack.unpack
+    elif url.endswith(".yaml"):
+        import yaml
+
+        mode = "r"
+        reader = yaml.safe_load
+    else:
+        raise ValueError(f"url ({url}) must have a .msgpack.gz or .yaml suffix")
+
+    with fsspec.open(url, mode, compression="infer") as f:
+        return reader(f)
+
+
+class DumpArtefact:
```

Comment:
I'm +1 on a wrapper class mostly because I think standard OO Encapsulation is the correct strategy to take for relating to the contents of this dump via `scheduler_tasks_in_state` or `scheduler_story`, for example. I personally find the idea of a collection of module functions for interacting with the `state` dict unpleasant -- I'd far rather have all the functionality for interacting with the dump in one place.

I do take your point about being able to poke around in the dump, so I've made `DumpArtefact` implement the `Mapping` ABC, and proxied the required `Mapping` implementations on `self.dumps`.